### ISSUE TITLE
 [Held requests] Approve modal does not appear when one of the expense is held

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#41655]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 
**Reproducible in staging?:** Needs reproduction
**Reproducible in production?:** Needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1714845351600779

## Deliverable
Provide reliable reproduction steps for the bug below. All assignees must be able to reproduce using the steps to be compensated.  

## Action Performed:
1. Go to staging.new.expensify.com
2. As an employee submit multiple expenses with scanned receipts
3. Login as approver and open the report
4. Put one scanned expense on hold
5. Navigate to another report and navigate back to the original report
6. Approve the report 

**NOTE:** step 5 is just to forcefully trigger OpenReport which resets the transaction's hold state. The video in OP did not do the same but somehow it might trigger OpenReport (when and where to call this API are quite unreliable and we're working on it).
 
## Expected Result:
Modal prompt to decide to approve the held or unheld total
## Actual Result:
No modal displayed and both expense got approved.
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [X] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/a0db747a-a37f-432c-9b99-13cf5cb1ade1

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @AndrewGable</details>

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by stephanieelliott** [comment]
I'm not able to reproduce this, adding the label to get it on the weekly retest rotation.

**Comment by puneetlath** [comment]
I experience it about 50% of the time when I try to approve with held expenses.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by stephanieelliott** [comment]
Still trying to repro, I think there is a proposal in the works to create jobs for these so will keep an eye there bc I do think this is worth fixing.

**Comment by mallenexpensify** [comment]
Updated title with prefix `[$125 Reproducible Steps]` 
Updated OP to inc. 
> ## Deliverable
> Provide reliable reproduction steps for the bug below. All assignees must be able to reproduce using the steps to be compensated.  

Using this as a test issue for
- https://github.com/Expensify/Expensify/issues/398280

**Comment by stephanieelliott** [comment]
^^ using this issue to test drive the [new process](https://expensify.slack.com/archives/C01SKUP7QR0/p1715892491306379) for `Create jobs for “Needs Reproduction” issues to document reproducible steps`.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by stephanieelliott** [comment]
Thanks @gijoe0295, @eh2077 can you review [these](https://github.com/Expensify/App/issues/41655#issuecomment-2123694055) proposal/repro steps when you get a sec?

**Comment by eh2077** [comment]
@gijoe0295 I have trouble to reproduce it. Can detail reproduce steps and share a recording please?

**Comment by eh2077** [comment]
@gijoe0295 Thank you so much for sharing the video! Yeah I can reproduce it and I also agree with you that it is caused by the unreliable API.

Cc @stephanieelliott @mallenexpensify

**Comment by eh2077** [comment]
@gijoe0295 Thank you so much for sharing the video! Yeah I can reproduce it and I also agree with you that it is caused by the unreliable API.

Reproducible steps: https://github.com/Expensify/App/issues/41655#issuecomment-2123694055
Video: https://github.com/Expensify/App/issues/41655#issuecomment-2133275564

Cc @stephanieelliott @mallenexpensify 

🎀👀🎀 C+ reviewed

**Comment by mallenexpensify** [comment]
oh.. sweet!  Thanks @gijoe0295 , we've been needing examples of `Reproducible Steps` issues. 
Once @AndrewGable chimes in about your proposals, I'll propose next steps for payment

**Comment by AndrewGable** [comment]
Hello I've taken a look, but I am a bit confused. Do I have to approve the reproducible steps then we can do a technical proposal?

**Comment by eh2077** [comment]
> Hello I've taken a look, but I am a bit confused. Do I have to approve the reproducible steps then we can do a technical proposal?

cc @mallenexpensify

**Comment by eh2077** [comment]
Not overdue, waiting for @mallenexpensify to drive clarity on this new flow, see https://github.com/Expensify/App/issues/41655#issuecomment-2142529255

**Comment by mallenexpensify** [comment]
Sorry @AndrewGable , I def could have been more clear.  And... after thinking about it more, I think the steps for confirming 'needs reproduction' issues and reproducible steps should be
1. Contributor submits steps
2. C+ confirms they're able to reproduce
3. BZ confirms then pays out, updated steps in OP, removes 'needs reproduction', removes job price, and ask the contributor who provided the steps if they want to submit a proposal to fix the issue before `Help Wanted` is added. 
4. If the contributor wants to submit a proposal they have ~48 hours to do so, then C+ reviews
5. If not, BZ adds `Help Wanted` and sets the job price for fixing it. 

Thanks for the patience here, this is a work-in-progress as we build out the [process for Needs Reproduction issues](https://github.com/Expensify/Expensify/issues/398280) (which should be wrapped up by the end of next week)

**Comment by stephanieelliott** [comment]
- BZ confirms then pays out 
   - Confirmed, I extended an offer to you @gijoe0295 here: https://www.upwork.com/nx/wm/offer/102582748
 - updated steps in OP, removes 'needs reproduction', removes job price
    - done 
 - and ask the contributor who provided the steps if they want to submit a proposal to fix the issue before Help Wanted is added.
    - @gijoe0295 see the above -- you have first dibs here, would you like to submit a proposal to fix this? 

Thanks for helping work this out @mallenexpensify! I have some ideas for how we can incorporate the repro job into our process a little more seamlessly, will bring that to [the thread](https://expensify.slack.com/archives/C01SKUP7QR0/p1715892491306379) in Slack!

**Comment by mallenexpensify** [comment]
@AndrewGable can you confirm this is BE ? If so, can you update the labels to `Internal`? Thx

**Comment by AndrewGable** [comment]
Yes, it seems backend related, but I am asking @puneetlath to see if he can still reproduce as we closed a related issue: https://github.com/Expensify/App/issues/40829

**Comment by puneetlath** [comment]
I'm still able to reproduce it sometimes, though not consistently.

**Comment by robertjchen** [comment]
Taking over on this one and checking it out

**Comment by robertjchen** [comment]
Saw it in action on production in Puneet's account yesterday. Seems to be a race condition bug where approval is allowed if the hold state has not taken "hold" yet.

Since the approve command defaults to full approval, the whole report is approved even if it was successfully held in the backend. Working on reproducing locally to diagnose further.

**Comment by robertjchen** [comment]
Had to unblock other Held items last week, diving into this this week.

**Comment by mallenexpensify** [comment]
Thanks for the help  Bobby, this one's been a pain to figure out.

**Comment by robertjchen** [comment]
Yep, good catch- it definitely seems like it 👍

**Comment by robertjchen** [comment]
Testing in local dev, aiming for PR by EOW

**Comment by robertjchen** [comment]
Evaluating if this can be addressed from the frontend 👀

**Comment by stephanieelliott** [comment]
Oop, thanks for the bump @gijoe0295! Paid in Upwork for the repro job and ended the contract.

**Comment by mallenexpensify** [comment]
Contributor: @gijoe0295 paid $125 via Upwork for reproducible steps. 
 
@stephanieelliott , added the above for consistency with documenting all payments.

**Comment by robertjchen** [comment]
diving into this first thing this week

**Comment by robertjchen** [comment]
Was sick last week, getting back into this.

**Comment by robertjchen** [comment]
Writing up a new approach, hang tight

**Comment by robertjchen** [comment]
Just getting to this now, update shortly

**Comment by robertjchen** [comment]
So, the reason we were holding off is that it's tangentially related to the optimistic changes from: https://github.com/Expensify/App/pull/42573

We encountered this during testing there and needed to get those changes out to better isolate the issue.

**Comment by robertjchen** [comment]
Getting to this one before EOW, was focused on clearing out the other held issues raised this week.

**Comment by robertjchen** [comment]
Started looking into this on the backend over the weekend, was unable to pinpoint the source as the flows don't reveal anything obvious that would cause this discrepancy. Investigation continues!

**Comment by robertjchen** [comment]
Reproduced again here: https://expensify.slack.com/archives/C05LX9D6E07/p1723143340548779

**Comment by robertjchen** [comment]
Potentially it would be a frontend fix- it seems like we're not optimistically updating that report/expense (preview?) such that we prevent immediate approval after the hold action

Though, I wonder if we wait long enough at the last step- eventually that preview becomes held? If so, that means the backend is okay and we can just focus on the frontend.

**Comment by robertjchen** [comment]
Confirming if it's [backend or frontend or both](https://expensify.slack.com/archives/C05LX9D6E07/p1723485013528539?thread_ts=1723143340.548779&cid=C05LX9D6E07)

**Comment by robertjchen** [comment]
[Slack Discussion here](https://expensify.slack.com/archives/C05LX9D6E07/p1723706879833979?thread_ts=1723143340.548779&cid=C05LX9D6E07). We have a backend fix being deployed for this!

**Comment by robertjchen** [comment]
The fix just went out! Let's re-test this

**Comment by puneetlath** [comment]
I confirmed it seems to be fixed for me. Let's make sure we get a new regression test for this scenario with scanning receipts @stephanieelliott

**Comment by robertjchen** [comment]
Awesome, any other remaining steps here before we can close this one out?

**Comment by stephanieelliott** [comment]
@isagoico created the QA test for us ❤️ https://github.com/Expensify/Expensify/issues/421472

Seems we're all set to close now!
